src/test/java/com/amazon/opendistroforelasticsearch/sql/esintgtest/TermQueryExplainIT.java

```diff
@@ -275,6 +278,65 @@ public void testJoinAliasMissing() throws IOException {
 
     }
 
+    @Test
+    public void testNestedSingleConditionAllFields() throws IOException {
+        String result = explainQuery(String.format(Locale.ROOT,
+            "SELECT * FROM %s e, e.projects p WHERE p.name = 'something' ",
+            TEST_INDEX_EMPLOYEE_NESTED));
+        assertThat(result, containsString("\"term\":{\"projects.name.keyword\":{\"value\":\"something\""));
+        assertThat(result, containsString("\"path\":\"projects\""));
+    }
+
+    @Test
+    public void testNestedMultipleCondition() throws IOException {
+        String result = explainQuery(String.format(Locale.ROOT,
+            "SELECT e.id, p.name FROM %s e, e.projects p WHERE p.name = 'something' and p.started_year = 1990 ",
+            TEST_INDEX_EMPLOYEE_NESTED));
+        assertThat(result, containsString("\"term\":{\"projects.name.keyword\":{\"value\":\"something\""));
+        assertThat(result, containsString("\"term\":{\"projects.started_year\":{\"value\":1990"));
+        assertThat(result, containsString("\"path\":\"projects\""));
+    }
+
+    @Test
+    public void testConditionsOnDifferentNestedDocs() throws IOException {
+        String result = explainQuery(String.format(Locale.ROOT,
+            "SELECT p.name, c.likes  FROM %s e, e.projects p, e.comments c WHERE p.name = 'something' or c.likes = 56 ",
+            TEST_INDEX_EMPLOYEE_NESTED));
+        assertThat(result, containsString("\"term\":{\"projects.name.keyword\":{\"value\":\"something\""));
+        assertThat(result, containsString("\"term\":{\"comments.likes\":{\"value\":56"));
+        assertThat(result, containsString("\"path\":\"projects\""));
+        assertThat(result, containsString("\"path\":\"comments\""));
+    }
+
+    @Test
+    public void testNestedSingleConditionSpecificFields() throws IOException {
+        String result = explainQuery(String.format(Locale.ROOT,
+            "SELECT e.id, p.name FROM %s e, e.projects p WHERE p.name = 'hello' or p.name = 'world' ",
+            TEST_INDEX_EMPLOYEE_NESTED));
+        assertThat(result, containsString("\"term\":{\"projects.name.keyword\":{\"value\":\"hello\""));
+        assertThat(result, containsString("\"term\":{\"projects.name.keyword\":{\"value\":\"world\""));
+        assertThat(result, containsString("\"path\":\"projects\""));
+    }
+
+    @Test
+    public void testNestedSingleGroupBy() throws IOException {
+        String result = explainQuery(String.format(Locale.ROOT,
+            "SELECT e.id, p.name FROM %s e, e.projects p GROUP BY  p.name ",
+            TEST_INDEX_EMPLOYEE_NESTED));
+        assertThat(result, containsString("\"terms\":{\"field\":\"projects.name.keyword\""));
+        assertThat(result, containsString("\"nested\":{\"path\":\"projects\""));
+    }
+
+    @Test
+    public void testNestedSingleOrderBy() throws IOException {
+        String result = explainQuery(String.format(Locale.ROOT,
```

Comment:
consider inlining index name and getting rid of string format entirely

Reply:
I am all for it, just wanted to follow the style of other tests in this testsuite .

Reply:
Done, made them all inline.

Reply:
np: I am a little skeptical about this one: it's fine not to use format, because the index names do not have any special characters which depend on the locale, but I still like keeping them in variables. In a case if we need to change the name, we would change in one place, instead of doing find and replace across the file. With static vars, just looking at the test, you can quickly identify which index(es) the test operates on because of the all caps naming, whereas with index names you need to scan through the query. Finally, since we have to have the index names in variables anyway (for initialization code), it naturally makes sense to re-use those variables.

Reply:
Keeping it inline for now.